src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultAssemblyPartDiscoveryProvider.cs

```diff
@@ -101,23 +100,28 @@ private Dependency CreateDependency(RuntimeLibrary library, ISet<string> referen
 
             private DependencyClassification ComputeClassification(string dependency)
             {
-                Debug.Assert(_dependencies.ContainsKey(dependency));
+                if (!_runtimeDependencies.ContainsKey(dependency))
```

Comment:
Was that assert not true? I don't understand how this is an issue. Has something changed in DependencyContext?